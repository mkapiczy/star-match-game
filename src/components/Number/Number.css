.number {
    margin-right: 5px;
    background-color: #aeaeae;
    border: none;
    color: white;
    text-align: center;
    text-decoration: none;
    border-radius: 50%;
    width: 5vmin;
    height: 5vmin;
}

.number.not-available {
    background-color: #3F51B5;
}

.number:not(.not-available):hover {
    background-color: #d7d7d7;
}