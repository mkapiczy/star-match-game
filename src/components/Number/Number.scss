.number {
  margin-right: 0.25em;
  margin-left: 0.25em;
  background-color: #aeaeae;
  border: none;
  color: white;
  text-align: center;
  text-decoration: none;
  border-radius: 50%;
  width: 5vmin;
  height: 5vmin;
  font-size: 2.5vmin;

  &.not-available {
    background-color: #3F51B5;
  }

  &:not(.not-available):hover {
    background-color: #d8d8d8;
  }
}