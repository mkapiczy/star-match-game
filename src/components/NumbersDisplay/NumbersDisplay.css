.NumbersDisplay {
    flex: 1 0 auto;
    background-color: gray;
}