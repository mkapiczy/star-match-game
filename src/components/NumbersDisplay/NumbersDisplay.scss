.NumbersDisplay {
    display: flex;
    flex-direction: row;
    flex: 1 0 auto;
    flex-wrap: wrap;
    height: 80%;
    background-color: white;
    justify-content: flex-start;
    align-items: center;
    border-radius: 2.5vmin;
    border-style: groove;
    margin: 2vmin;
    padding: 1vmin;
}
