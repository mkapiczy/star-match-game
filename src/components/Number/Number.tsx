import React from 'react';
import './Number.scss';
import classNames from "classnames";

interface Props {
    numberValue: number,
    onNumberClick: (numberValue: number) => void;
    isAvailable: boolean
}

const Number: React.FC<Props> = (props: Props) => {
    return (
        <button key={props.numberValue} className={classNames('number', {'not-available': !props.isAvailable})}
                disabled={!props.isAvailable} onClick={() => props.onNumberClick(props.numberValue)}>
            {props.numberValue}
        </button>
    )
}

export default Number;