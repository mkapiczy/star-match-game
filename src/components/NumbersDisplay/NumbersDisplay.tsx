import React from 'react';
import './NumbersDisplay.css';

interface Props {
    availableNumbers: Array<number>,
    onNumberClick: (selectedNumber: number) => void;
}

const NumbersDisplay: React.FC<Props> = (props: Props) => {
    return (
        <div className="NumbersDisplay">
            {[...Array(9)].map((e, i) => <button
                className={'circular-button' + (props.availableNumbers.includes(i + 1) ? '' : ' not-available')} onClick={() => props.onNumberClick(i+1)}>{i + 1}</button>)}
        </div>
    )
}

export default NumbersDisplay;