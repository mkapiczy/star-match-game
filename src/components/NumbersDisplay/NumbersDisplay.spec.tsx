import React from 'react';
import NumbersDisplay from "./NumbersDisplay";
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import Number from "../Number/Number";

describe("NumbersDisplay component snapshot tests", () => {
    it("should match snapshot for all numbers available", () => {
        const component = renderer.create(
            <NumbersDisplay numbersCount={9} onNumberClick={() => ''} availableNumbers={[1, 2, 3, 4, 5, 6, 7, 8, 9]}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("should match snapshot for no numbers available", () => {
        const component = renderer.create(
            <NumbersDisplay numbersCount={9} onNumberClick={() => ''} availableNumbers={[]}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("should match snapshot for some numbers available", () => {
        const component = renderer.create(
            <NumbersDisplay numbersCount={9} onNumberClick={() => ''} availableNumbers={[1, 2, 7, 8]}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('NumbersDisplay component tests', () => {
    it("should render correct number of Number components", () => {
        const expectedNumberOfNumberComponents = 9
        const wrapper = mount(<NumbersDisplay numbersCount={expectedNumberOfNumberComponents} availableNumbers={[]}
                                              onNumberClick={() => ''}/>)
        const renderedNumberComponents = wrapper.find(Number)
        expect(renderedNumberComponents.length).toBe(expectedNumberOfNumberComponents)
    })
    it("should render all available numbers as available with correct numberValue", () => {
        const expectedNumberOfNumberComponents = 9
        const availableNumbers = [1, 2, 3, 4, 5, 6, 7, 8, 9]
        const wrapper = mount(<NumbersDisplay numbersCount={expectedNumberOfNumberComponents} availableNumbers={availableNumbers}
                                              onNumberClick={() => ''}/>)
        const renderedNumberComponents = wrapper.find(Number)
        renderedNumberComponents.forEach((component, i) => {
            expect(component.prop("isAvailable")).toBe(true)
            expect(component.prop("numberValue")).toBe(i + 1)
        })
    })

    it("should render all numbers as not available with correct numberValue", () => {
        const expectedNumberOfNumberComponents = 9
        const wrapper = mount(<NumbersDisplay numbersCount={expectedNumberOfNumberComponents} availableNumbers={[]}
                                              onNumberClick={() => ''}/>)
        const renderedNumberComponents = wrapper.find(Number)
        renderedNumberComponents.forEach((component, i) => {
            expect(component.prop("isAvailable")).toBe(false)
            expect(component.prop("numberValue")).toBe(i + 1)
        })
    })

    it("should call onNumberClick function when click triggered on Number with isAvailable true", () => {
        const onNumberClick = jest.fn()
        const expectedNumberOfNumberComponents = 9
        const availableNumbers = [1]
        const wrapper = mount(<NumbersDisplay numbersCount={expectedNumberOfNumberComponents} availableNumbers={availableNumbers}
                                              onNumberClick={onNumberClick}/>)
        const renderedNumberComponents = wrapper.find(Number)
        const availableNumber = renderedNumberComponents.find({"isAvailable": true}).first()
        availableNumber.simulate("click")
        expect(onNumberClick).toBeCalledTimes(1)
    })

    it("should not call onNumberClick function when click triggered on Number with isAvailable false", () => {
        const onNumberClick = jest.fn()
        const expectedNumberOfNumberComponents = 9
        const availableNumbers = [1]
        const wrapper = mount(<NumbersDisplay numbersCount={expectedNumberOfNumberComponents} availableNumbers={availableNumbers}
                                              onNumberClick={onNumberClick}/>)
        const renderedNumberComponents = wrapper.find(Number)
        const availableNumber = renderedNumberComponents.find({"isAvailable": false}).first()
        availableNumber.simulate("click")
        expect(onNumberClick).not.toBeCalled()
    })
});