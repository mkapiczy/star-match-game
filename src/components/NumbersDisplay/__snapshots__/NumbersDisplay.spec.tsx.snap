// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersDisplay component snapshot tests should match snapshot for all numbers available 1`] = `
<div
  className="NumbersDisplay"
>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    9
  </button>
</div>
`;

exports[`NumbersDisplay component snapshot tests should match snapshot for no numbers available 1`] = `
<div
  className="NumbersDisplay"
>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    9
  </button>
</div>
`;

exports[`NumbersDisplay component snapshot tests should match snapshot for some numbers available 1`] = `
<div
  className="NumbersDisplay"
>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    9
  </button>
</div>
`;

exports[`NumbersDisplay component tests should match snapshot for all numbers available 1`] = `
<div
  className="NumbersDisplay"
>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    9
  </button>
</div>
`;

exports[`NumbersDisplay component tests should match snapshot for no numbers available 1`] = `
<div
  className="NumbersDisplay"
>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    9
  </button>
</div>
`;

exports[`NumbersDisplay component tests should match snapshot for some numbers available 1`] = `
<div
  className="NumbersDisplay"
>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="number"
    disabled={false}
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="number not-available"
    disabled={true}
    onClick={[Function]}
  >
    9
  </button>
</div>
`;
